dll/cpl/desk/settings.c

```diff
@@ -866,10 +899,9 @@ ApplyDisplaySettings(HWND hwndDlg, PSETTINGS_DATA pData)
     }
     else
     {
-        pData->CurrentDisplayDevice->CurrentSettings->dmPelsWidth = pData->CurrentDisplayDevice->InitialSettings.dmPelsWidth;
-        pData->CurrentDisplayDevice->CurrentSettings->dmPelsHeight = pData->CurrentDisplayDevice->InitialSettings.dmPelsHeight;
-        pData->CurrentDisplayDevice->CurrentSettings->dmBitsPerPel = pData->CurrentDisplayDevice->InitialSettings.dmBitsPerPel;
-        pData->CurrentDisplayDevice->CurrentSettings->dmDisplayFrequency = pData->CurrentDisplayDevice->InitialSettings.dmDisplayFrequency;
+        PSETTINGS_ENTRY pInitialSettings = &pData->CurrentDisplayDevice->InitialSettings;
+        pData->CurrentDisplayDevice->CurrentSettings =
+            FindBestElement(pInitialSettings, pData->CurrentDisplayDevice->Settings);
```

Comment:
```suggestion
        pData->CurrentDisplayDevice->CurrentSettings =
            FindBestElement(&pData->CurrentDisplayDevice->InitialSettings,
                            pData->CurrentDisplayDevice->Settings);
```

--

Going a step further, you could pass `pData->CurrentDisplayDevice` only.

Reply:
Thanks.